stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/service/sync/OfflineSyncFirebaseMessagingService.kt

```diff
@@ -1,137 +1,19 @@
 package io.getstream.chat.android.livedata.service.sync
 
-import android.app.NotificationChannel
-import android.app.NotificationManager
-import android.content.Context
-import android.os.Build
-import androidx.core.app.NotificationCompat
 import com.google.firebase.messaging.FirebaseMessagingService
 import com.google.firebase.messaging.RemoteMessage
-import io.getstream.chat.android.client.ChatClient
-import io.getstream.chat.android.client.logger.ChatLogger
-import io.getstream.chat.android.client.models.User
-import io.getstream.chat.android.client.notifications.FirebaseMessageParser
-import io.getstream.chat.android.client.notifications.FirebaseMessageParserImpl
-import io.getstream.chat.android.client.notifications.handler.ChatNotificationHandler
-import io.getstream.chat.android.client.notifications.handler.NotificationConfig
-import io.getstream.chat.android.core.internal.coroutines.DispatcherProvider
-import io.getstream.chat.android.livedata.ChatDomain
-import kotlinx.coroutines.GlobalScope
-import kotlinx.coroutines.launch
 
 internal class OfflineSyncFirebaseMessagingService : FirebaseMessagingService() {
 
-    private val logger = ChatLogger.get("OfflineSyncFirebaseMessagingService")
-
-    private val syncModule by lazy { SyncProvider(this) }
-    private val notificationConfig: NotificationConfig by lazy { syncModule.notificationConfigStore.get() }
-    private val firebaseMessageParser: FirebaseMessageParser by lazy { FirebaseMessageParserImpl(notificationConfig) }
+    private val syncServiceDelegate: PushMessageSyncHandler =
```

Comment:
We still have the "delegate" naming in this property, also in some of the documentation comments added on this PR.